db/sqlinterfaces.c

```diff
@@ -3248,8 +3248,10 @@ static int put_prepared_stmt_int(struct sqlthdstate *thd,
         goto cleanup;
     }
     if (rec->stmt_entry != NULL) { /* we found this stmt in the cache */
-        /* Leave the ownership of stmt data. */
-        query_data_func(clnt, NULL, NULL, QUERY_STMT_DATA, QUERY_DATA_SET);
+        if (rec->status & CACHE_HAS_HINT) {
```

Comment:
this is caching enabled for all statements? why won't you want to give the ownership back in that case?

Reply:
Talked to Rivers on IB, we don't take params in case of non-hint statements.. so we shouldn't even care about them here.. so this is correct.

Reply:
Thank you Mohit!